wolfcrypt/src/evp.c

```diff
@@ -6878,6 +6878,446 @@ int wolfSSL_EVP_get_hashinfo(const WOLFSSL_EVP_MD* evp,
 }
 #endif /* !defined(NO_PWDBASED) */
 
+/* Base64 encoding APIs */
+#if defined(WOLFSSL_BASE64_ENCODE) || defined(WOLFSSL_BASE64_DECODE)
+static struct WOLFSSL_EVP_ENCODE_CTX* wolfSSL_EVP_ENCODE_CTX_new_ex(void* heap);
+/*  wolfSSL_EVP_ENCODE_CTX_new allocates WOLFSSL_EVP_ENCODE_CTX
+ *  Returns WOLFSSL_EVP_ENCODE_CTX structure on success, NULL on failure.
+ */
+struct WOLFSSL_EVP_ENCODE_CTX* wolfSSL_EVP_ENCODE_CTX_new(void)
+{
+    return wolfSSL_EVP_ENCODE_CTX_new_ex(NULL);
+}
+/* wolfSSL_EVP_ENCODE_CTX_new_ex is a helper function of
+ * wolfSSL_EVP_ENCODE_CTX_new to allocate WOLFSSL_EVP_ENCODE_CTX structure.
+ */
+static struct WOLFSSL_EVP_ENCODE_CTX* wolfSSL_EVP_ENCODE_CTX_new_ex(void* heap)
+{
+    WOLFSSL_EVP_ENCODE_CTX* ctx = NULL;
+    WOLFSSL_ENTER("wolfSSL_EVP_ENCODE_CTX_new");
+    ctx = (WOLFSSL_EVP_ENCODE_CTX*)XMALLOC( sizeof(WOLFSSL_EVP_ENCODE_CTX),heap,DYNAMIC_TYPE_OPENSSL );
+
+    if(ctx != NULL) {
+        XMEMSET(ctx,0,sizeof(WOLFSSL_EVP_ENCODE_CTX) );
+        ctx->heap = heap;
+        return ctx;
+    }
+    return NULL;
+}
+/*  wolfSSL_EVP_ENCODE_CTX_free frees specified WOLFSSL_EVP_ENCODE_CTX struc.
+ */
+void wolfSSL_EVP_ENCODE_CTX_free(WOLFSSL_EVP_ENCODE_CTX* ctx)
+{
+    WOLFSSL_ENTER("wolfSSL_EVP_ENCODE_CTX_free");
+    if(ctx != NULL) {
+        XFREE(ctx,ctx->heap,DYNAMIC_TYPE_OPENSSL);
+    }
+}
+#endif /* WOLFSSL_BASE64_ENCODE || WOLFSSL_BASE64_DECODE */
+#if defined(WOLFSSL_BASE64_ENCODE)
+/*  wolfSSL_EVP_EncodeInit initializes specified WOLFSSL_EVP_ENCODE_CTX ojbect
+ *  for the subsequent wolfSSL_EVP_EncodeUpdate.
+ */
+void wolfSSL_EVP_EncodeInit(WOLFSSL_EVP_ENCODE_CTX* ctx)
+{
+    WOLFSSL_ENTER("wolfSSL_EVP_EncodeInit");
+
+    /* clean up ctx */
+    if(ctx != NULL) {
+        ctx->remaining = 0;
+        XMEMSET(ctx->data,0, sizeof(ctx->data));
+    }
+}
+/*  wolfSSL_EVP_EncodeUpdate encodes the input data in 64-byte units
+ *  and outputs it to out. If less than 64 bytes of data remain, save it in
+ *  ctx. The data given in the subsequent wolfSSL_EVP_EncodeUpdate
+ *  is combined with the data stored in CTX and used for encoding.
+ *  Returns 1 on success, 0 on error.
+ */
+int  wolfSSL_EVP_EncodeUpdate(WOLFSSL_EVP_ENCODE_CTX* ctx,
+                unsigned char*out, int *outl, const unsigned char*in, int inl)
+{
+    int cpysz;
+    int res;
+    word32 outsz = 0;
+
+    WOLFSSL_ENTER("wolfSSL_EVP_EncodeUpdate");
+
+    if( ctx == NULL || out == NULL || in == NULL || outl == NULL )
+        return 0;
+
+    *outl = 0;
+
+    /* if the remaining data exit in the ctx, add input data to them to create a block(48bytes) for encoding*/
```

Comment:
Wrap lines to multiple lines if longer than 80 characters, to match our coding standard requirements.  Apply to rest of PR as well, thanks.

Reply:
Fixed it.